### PR TITLE
[tftrt]

### PR DESCRIPTION
**Pull request by jjsjann123**
trt 4 update input check on tensor dimensions.
  relaxing tensor dimension for trt 4 to support non-4-dimensional inputs
  added unit test (currently disabled due to failed INT8 conversion)

### CODE REVIEW

File: tensorflow/contrib/tensorrt/convert/convert_nodes.cc [hunk 1]
```diff
@@ -2693,8 +2693,13 @@ tensorflow::Status ConvertGraphDefToEngine(
     VLOG(1) << "Converting op name=" << node_name << ", op=" << node_def.op();
     if (tensorflow::str_util::StartsWith(node_name, kInputPHName) &&
         (node_def.op() == "Placeholder")) {
-      nvinfer1::DimsCHW input_dim_pseudo_chw;
-      for (int i = 0; i < 8; i++) input_dim_pseudo_chw.d[i] = 0;
+      nvinfer1::DataType dtype(nvinfer1::DataType::kFLOAT);
```
Comment (aaroey) [inline tensorflow/contrib/tensorrt/convert/convert_nodes.cc]: This conversion is already done in ValidateInputProperties(), I guess you may need to sync with master to merge to changes from PR #20755.

File: tensorflow/contrib/tensorrt/convert/convert_nodes.cc [hunk 2]
```diff
@@ -2721,13 +2726,27 @@ tensorflow::Status ConvertGraphDefToEngine(
         StrAppend(&dim_str, " ]");
         VLOG(1) << dim_str;
       }
+
+#if NV_TENSORRT_MAJOR == 3
+      nvinfer1::DimsCHW input_dim;
```
Comment (aaroey) [inline tensorflow/contrib/tensorrt/convert/convert_nodes.cc]: Would you please put these inside ValidateInputeProperties()?

Comment (jjsjann123) [reply tensorflow/contrib/tensorrt/convert/convert_nodes.cc]: A second thought, we should actually put it in InputEdgeValidator. As this check only applies to live-in tensors

**Comment by aaroey** [comment]
Thanks @jjsjann123. I'll wait for #21138 before merging this one.

**Comment by aaroey** [comment]
Hi @jjsjann123, would help to merge the conflicts? I'll test and merge after that.

**Comment by aaroey** [comment]
As chatted offline, I'll try to extend this PR to fix all commented tests, by either minimizing the number of disabled test cases, or just fixing them.

**Comment by jjsjann123** [comment]
sync @pooyadavoodi since he's also looking at fixing unit tests

**Review by aaroey — APPROVED** [review]

**Pull request merged**